### PR TITLE
build: Replace generic issue templates with issue forms

### PR DESCRIPTION
**Pull request by fghaas**
Convert the remaining pure Markdown issue report templates to forms.

Also, add automatic issue labels.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/textual_bug.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        Quote the passage that you believe needs improvement.
+      label: What the documentation says
+    id: actual
+    type: textarea
+    validations:
+      required: true
+  - attributes:
+      description: >
+        Make a suggestion on how to improve the passage and to make it
+        more understandable, more accessible, or simply more correct.
```
Comment (colder-is-better) [inline .github/ISSUE_TEMPLATE/textual_bug.yml:38]:
```suggestion
        more understandable, more accessible, or simply better.
```

File: .github/ISSUE_TEMPLATE/functional_bug.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        it here. Otherwise, you can leave this blank.
+      label: Additional context
+    id: context
+    type: textarea
+    validations:
+      required: false
+description: >
+  Use this template to report functionally incorrect or misleading
+  documentation.
+labels: functional-bug
```
Comment (colder-is-better) [inline .github/ISSUE_TEMPLATE/functional_bug.yml:88]:
```suggestion
label: functional-bug
```

File: .github/ISSUE_TEMPLATE/gap.yml [hunk 3]
```diff
@@ -57,4 +57,5 @@ body:
 description: >
   Use this template to report documentation gaps: information that is
   missing altogether, or that is incomplete.
+labels: gap
```
Comment (colder-is-better) [inline .github/ISSUE_TEMPLATE/gap.yml:60]:
```suggestion
label: gap
```

File: .github/ISSUE_TEMPLATE/textual_bug.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        If there’s any other context you’d like to share, please add
+        it here. Otherwise, you can leave this blank.
+      label: Additional context
+    id: context
+    type: textarea
+    validations:
+      required: false
+description: >
+  Use this template to report grammar glitches, syntax errors, and typos.
+labels: textual-bug
```
Comment (colder-is-better) [inline .github/ISSUE_TEMPLATE/textual_bug.yml:55]:
```suggestion
label: textual-bug
```

File: .github/ISSUE_TEMPLATE/functional_bug.yml (hunk 2 above)
Comment (fghaas) [reply .github/ISSUE_TEMPLATE/functional_bug.yml:88]: What, you think we get to make up our own syntax? :)

I'm afraid we'll have to follow [what GitHub gives us](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#top-level-syntax).

File: .github/ISSUE_TEMPLATE/textual_bug.yml (hunk 4 above)
Comment (fghaas) [reply .github/ISSUE_TEMPLATE/textual_bug.yml:55]: [Nope](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#top-level-syntax).

File: .github/ISSUE_TEMPLATE/textual_bug.yml (hunk 1 above)
Comment (fghaas) [reply .github/ISSUE_TEMPLATE/textual_bug.yml:38]: No, I consciously used "more correct" here.

File: .github/ISSUE_TEMPLATE/gap.yml (hunk 3 above)
Comment (fghaas) [reply .github/ISSUE_TEMPLATE/gap.yml:60]: Again, [nope](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#top-level-syntax), sorry. :)

File: .github/ISSUE_TEMPLATE/functional_bug.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
+# yamllint disable rule:line-length
+---
+body:
+  - attributes:
+      description: >-
```
Comment (colder-is-better) [inline .github/ISSUE_TEMPLATE/functional_bug.yml:5]: I see that most of the time you use `>` for multiline strings, but there are some times you use `>-` (which means "don't append a linebreak at the end", if I'm not mistaken). Is there a reason for that?

**Review by colder-is-better — COMMENTED** [review]
All is looking good -- I only have some pico-nits and a question.

File: .github/ISSUE_TEMPLATE/functional_bug.yml (hunk 5 above)
Comment (fghaas) [reply .github/ISSUE_TEMPLATE/functional_bug.yml:5]: Not really. Emacs just automatically turns `>` into `>-`.

But GitHub's issue forms renderer doesn't even distinguish between `>` (multiline string, drop newlines) and `|` (multiline string, retain newlines); it always behaves as if using `|`.

**Review by colder-is-better — APPROVED** [review]

**Pull request merged**